app/models/candidate_interface/degrees_form.rb

```diff
@@ -15,17 +15,51 @@ class DegreesForm
 
     validate :award_year_is_date, if: :award_year
 
-    def self.build_from_application(application_form)
-      application_form.application_qualifications.degrees.map do |degree|
+    class << self
+      def build_from_application(application_form)
+        application_form.application_qualifications.degrees.map do |degree|
+          new(
+            id: degree.id,
+            qualification_type: degree.qualification_type,
+            subject: degree.subject,
+            institution_name: degree.institution_name,
+            grade: degree.grade,
+            predicted_grade: degree.predicted_grade,
+            award_year: degree.award_year,
+          )
+        end
+      end
+
+      def find_using_application(application_form, degree_id)
+        degree = application_form.application_qualifications.find(degree_id)
+        grade = determine_application_grade(degree.grade, degree.predicted_grade)
+
         new(
+          id: degree.id,
           qualification_type: degree.qualification_type,
           subject: degree.subject,
           institution_name: degree.institution_name,
-          grade: degree.grade,
-          predicted_grade: degree.predicted_grade,
+          grade: grade,
+          other_grade: grade == 'other' ? degree.grade : '',
+          predicted_grade: degree.predicted_grade ? degree.grade : '',
           award_year: degree.award_year,
         )
       end
+
+    private
+
+      def determine_application_grade(grade, predicted_grade)
+        case grade
+        when 'first', 'upper_second', 'lower_second', 'third'
+          grade
+        else
+          if predicted_grade
+            'predicted'
+          else
+            'other'
+          end
+        end
+      end
```

Comment:
it's nice that you used a private method for this logic,
but it sounds it should not be in an obj whose purpose is to create a form for Degree.

It might be better to have it in the model: `degree.get_grade` 
(it is still not the best place, but I think it's more appropriate)